qhana_plugin_runner/api/tasks_api.py

```diff
@@ -80,35 +80,34 @@ def remove_empty_attributes(self, data: Dict[str, Any], **kwargs):
             del data["outputs"]
         if data["steps"] == None:
             del data["steps"]
+        if data["progress"] == None:
             del data["progress"]
         return data
 
 
-@TASKS_API.route("/<string:db_id>/")
+@TASKS_API.route("/<string:task_id>/")
 class TaskView(MethodView):
     """Task status resource."""
 
     @TASKS_API.response(HTTPStatus.OK, TaskStatusSchema())
-    def get(self, db_id: str):
+    def get(self, task_id: str):
         """Get the current task status."""
-        task_data: Optional[ProcessingTask] = ProcessingTask.get_by_id(id_=db_id)
+        task_data: Optional[ProcessingTask] = ProcessingTask.get_by_id(id_=task_id)
         if task_data is None:
             abort(HTTPStatus.NOT_FOUND, message="Task not found.")
             return  # return for type checker, abort raises exception
 
-        assert (
-            task_data.task_id is not None
-        ), "If this is None then get_task_by_id is faulty."  # assertion for type checker
-
         progress = None
-        steps = None
-        if task_data.multi_step:
-            progress = {
+        if task_data.progress_value:
+            progress = progress = {
                 "value": task_data.progress_value,
                 "start": task_data.progress_start,
                 "target": task_data.progress_target,
                 "unit": task_data.progress_unit,
             }
+
+        steps = None
+        if task_data.multi_step:
```

Comment:
why the extra boolean flag for multi step? Is it not sufficient to check the length of the step list as fetched from DB?